packages/mdc-icon-button/_mixins.scss

```diff
@@ -87,6 +87,10 @@
     height: $height + $padding * 2;
     padding: $padding;
     font-size: max($width, $height);
+
+    &.mdc-text-field__icon {
```

Comment:
This is applying the styles based on the context. It should be other way around. That is, parent component needs to apply styles for sub component.

Also, I feel the leading / trailing icon should be center aligned no matter the size of the icon.

For example,

```css
.mdc-text-field__icon {
  ...
  top: 50%;
  transform: translateY(-50%);
}
```

But we also need to consider adding enough (left / right based on the position of icon) padding to input element to avoid input overlapping with icon.

Reply:
> This is applying the styles based on the context. It should be other way around. That is, parent component needs to apply styles for sub component.
This is what i wanted to do initially. But because i chose to fix the issue using negative margin, i needed the padding value from the icon button and switched things around. Should I 

> Also, I feel the leading / trailing icon should be center aligned no matter the size of the icon.
Isn't the size of the icon strictly defined in the spec?

> But we also need to consider adding enough (left / right based on the position of icon) padding to input element to avoid input overlapping with icon.
That's something i am looking at

Reply:
IMO, it is not safe to assume if the icon size will be fixed. For example, in future icon button may introduce dense variant which is smaller than standard variant.